submit_reports/views.py

```diff
@@ -112,6 +111,47 @@ def faculty_view(request):
 	context = Context({'form': form,})
 	return HttpResponse(template.render(context=context, request=request))
 
+#View for TA
+########################################################################## 
+from django.template.backends.utils import csrf_input_lazy, csrf_token_lazy
+@login_required
+@user_passes_test(lambda u: u.is_superuser or u.faculty is not None)
+def ta_view(request):
+	
+	reports = SubmitReport.objects.query_pending_reports()
+	reports = reports.filter(courses__in=request.user.staff.courses.all()).distinct()
+
+	form = ReportSearchForm(request.POST)
+	courses = request.user.faculty.course_set.all()
+	df = pd.DataFrame(list(reports.values(
+		'first_name', 'last_name', 'start_time', 'end_time', 'summary')))
+	from django.template import Template, Context
+	if form.is_valid():
+		reports = form.filter_queryset(request, reports)
+		df = pd.DataFrame(list(reports.values(
+		'first_name', 'last_name', 'start_time', 'end_time', 'summary')))
+	if reports:
+		table = df.to_html(escape=False, index=False,
+		columns=['first_name', 'last_name', 'start_time', 'end_time', 'summary'],
+		formatters={
+			'summary': (lambda s: '<abbr title=\"' + s + '\">Summary</abbr>'),
+			'submitter': (lambda s: Student.objects.get(pk=s).__unicode__()),
+			'start_time': (lambda s: readable_datetime(s)),
+			'end_time': (lambda s: readable_datetime(s)),
+		})
+	else:
+		table = "No reports matched your search."
+
+	temp = Template("""<form method='POST' action=''>
+		{% csrf_token %}
+		{{form.as_p}}
+		<input type='submit' value="Search", action="">
+		</form>"""
+		+ '\n' + table)
+	template = Tplate(temp)
+	context = Context({'form': form,})
+	return HttpResponse(template.render(context=context, request=request))
```

Comment:
We should probably look into abstracting this code away. @msuk @vogel-sa do you guys have any ideas?